src/python/zquantum/core/wip/circuit/_serde.py

```diff
@@ -0,0 +1,241 @@
+from typing import List, Iterable
+from functools import singledispatch
+
+import sympy
+
+from . import _gates as g
+from . import _builtin_gates as bg
+from ...utils import SCHEMA_VERSION
+
+
+def serialize_expr(expr):
```

Comment:
It wasn't immediately obvious to me that it's about sympy expressions.
Typing or changing name would help.